Uplift/Views/DropdownView.swift

```diff
@@ -0,0 +1,190 @@
+}
+
+class DropdownView: UIView {
+    var collapseHalfDropdownGesture: UITapGestureRecognizer!
+    var contentView: UIView!
+    var contentViewBottomConstraint: NSLayoutConstraint!
+    var contentViewHeight: CGFloat!
+    var contentViewHeightConstraint: NSLayoutConstraint!
+    var currentHeight: CGFloat!
+    var delegate: DropdownViewDelegate!
+    var expandCollapseDropdownGesture: UITapGestureRecognizer!
+    var expandFromHalfDropdownGesture: UITapGestureRecognizer!
+    var halfCollapseView: UIView?
+    var halfCollapseViewHeight: CGFloat!
+    var halfDropdownEnabled: Bool!
+    var halfExpandView: UIView?
+    var halfExpandViewHeight: CGFloat!
+    var halfHeight: CGFloat!
+    var headerView: UIView!
+    var headerViewHeight: CGFloat!
+    var status: DropdownStatus = .up
+    
+    init(headerView: UIView, headerViewHeight: CGFloat, contentView: UIView, contentViewHeight: CGFloat, halfDropdownEnabled: Bool = false, halfExpandView: UIView? = nil, halfExpandViewHeight: CGFloat = 0, halfCollapseView: UIView? = nil, halfCollapseViewHeight: CGFloat = 0, halfHeight: CGFloat = 0) {
+        super.init(frame: .zero)
+        
+        self.headerView = headerView
+        self.headerView.isUserInteractionEnabled = true
+        self.headerView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(self.headerView)
+        
+        self.headerViewHeight = headerViewHeight
+        currentHeight = headerViewHeight
+        
+        expandCollapseDropdownGesture = UITapGestureRecognizer(target: self, action: #selector(self.expandCollapseDropdown(sender:) ))
+        self.headerView.addGestureRecognizer(expandCollapseDropdownGesture)
+        
+        self.contentView = contentView
+        self.contentView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(self.contentView)
+        
+        currentHeight = headerView.bounds.height
+        
+        self.contentViewHeight = contentViewHeight
+        
+        self.halfDropdownEnabled = halfDropdownEnabled
+        
+        if let halfView = halfExpandView {
+            self.halfExpandView = halfView
+            self.halfExpandView!.translatesAutoresizingMaskIntoConstraints = false
+            self.halfExpandView!.isHidden = true
+            self.halfExpandView?.isUserInteractionEnabled = true
+            addSubview(self.halfExpandView!)
+            expandFromHalfDropdownGesture = UITapGestureRecognizer(target: self, action: #selector(self.expandFromHalfDropdown(sender:) ))
+            halfView.addGestureRecognizer(expandFromHalfDropdownGesture)
+        }
+        
+        self.halfExpandViewHeight = halfExpandViewHeight
+        
+        if let halfView = halfCollapseView {
+            self.halfCollapseView = halfView
+            self.halfCollapseView!.isHidden = true
+            self.halfCollapseView!.isUserInteractionEnabled = true
+            self.halfCollapseView!.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
Move this between lines 83 and 84 to match the ordering consistent with the if above